pennylane/tape/qscript.py

```diff
@@ -0,0 +1,1207 @@
+    so those arguments accept any iterable object:
+
+    >>> qscript = QuantumScript((qml.PauliX(i) for i in range(3)))
+    >>> qscript.circuit
+    [PauliX(wires=[0]), PauliX(wires=[1]), PauliX(wires=[2])]
+
+    """
+
+    def __init__(self, ops=None, measurements=None, prep=None, name=None, _update=True):
+        # are we going to queue these?
+        self.name = name
+        self._prep = [] if prep is None else list(prep)
+        self._ops = [] if ops is None else list(ops)
+        self._measurements = [] if measurements is None else list(measurements)
+
+        self._par_info = {}
+        """dict[int, dict[str, Operator or int]]: Parameter information. Keys are
+        parameter indices (in the order they appear on the quantum script), and values are a
+        dictionary containing the corresponding operation and operation parameter index."""
+
+        self._trainable_params = []
+        self._graph = None
+        self._specs = None
+        self._output_dim = 0
+        self._batch_size = None
+        self._qfunc_output = None
+
+        self.wires = _empty_wires
+        self.num_wires = 0
+
+        self.is_sampled = False
+        self.all_sampled = False
+
+        self._obs_sharing_wires = []
+        """list[.Observable]: subset of the observables that share wires with another observable,
+        i.e., that do not have their own unique set of wires."""
+        self._obs_sharing_wires_id = []
+
+        if _update:
+            self._update()
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__}: wires={self.wires.tolist()}, params={self.num_params}>"
+
+    @property
+    def hash(self):
+        """int: returns an integer hash uniquely representing the quantum script"""
+        fingerprint = []
+        fingerprint.extend(op.hash for op in self.operations)
+        fingerprint.extend(m.hash for m in self.measurements)
+        fingerprint.extend(self.trainable_params)
+        return hash(tuple(fingerprint))
+
+    def __iter__(self):
+        """list[.Operator, .MeasurementProcess]: Return an iterator to the
+        underlying quantum circuit object."""
+        return iter(self.circuit)
+
+    def __getitem__(self, idx):
+        """list[.Operator]: Return the indexed operator from underlying quantum
+        circuit object."""
+        return self.circuit[idx]
+
+    def __len__(self):
+        """int: Return the number of operations and measurements in the
+        underlying quantum circuit object."""
+        return len(self.circuit)
+
+    # ========================================================
+    # QSCRIPT properties
+    # ========================================================
+
+    @property
+    def interface(self):
+        """str, None: automatic differentiation interface used by the quantum script (if any)"""
+        return None
+
+    @property
+    def circuit(self):
+        """Returns the underlying quantum circuit object.
```

Comment:
Not sure about `quantum circuit object` - no such object.
```suggestion
        """Returns the underlying quantum circuit as a list of operations and measurements.
```